src/analysis/retail/priest/holy/modules/core/EchoOfLightAttributor.tsx

```diff
@@ -0,0 +1,128 @@
+import Analyzer from 'parser/core/Analyzer';
+import { GetRelatedEvents, HealEvent } from 'parser/core/Events';
+import { Options } from 'parser/core/Module';
+import Combatants from 'parser/shared/modules/Combatants';
+import { TALENTS_PRIEST } from 'common/TALENTS';
+import { ECHO_OF_LIGHT_ATTRIB_EVENT } from '../../normalizers/CastLinkNormalizer';
+import { ABILITIES_THAT_DONT_TRIGGER_MASTERY } from '../../constants';
+import StatTracker from 'parser/shared/modules/StatTracker';
+
+//TODO: move this to constants during planned spec overhaul
+
+const PRISMATIC_ECHOES_PER_RANK = 0.06;
+
+class EOLAttrib extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+    statTracker: StatTracker,
+  };
+
+  protected combatants!: Combatants;
+  protected statTracker!: StatTracker;
+  private masteryScaler = 1;
+
+  constructor(options: Options) {
+    super(options);
+    this.active = true;
+    if (this.selectedCombatant.hasTalent(TALENTS_PRIEST.PRISMATIC_ECHOES_TALENT)) {
+      const talentRank = this.selectedCombatant.getTalentRank(
+        TALENTS_PRIEST.PRISMATIC_ECHOES_TALENT,
+      );
+      this.masteryScaler *= talentRank * PRISMATIC_ECHOES_PER_RANK + 1;
+    }
+  }
+
+  /** Attributor for a spell's EOL amount,
+   *  it is calculated by looking at up to the next 6s of EOL ticks after a heal event
+   *  to get average overhealing, then it is raw healing event * mastery % * scaled overhealing
+   *
+   *  ASSUMPTION: whether or not the hot gets infinitely extended or pool empties within 2 ticks
+   *  won't have a noticeable impact on the value since any extensions would be a smaller and smaller value
+   *  of the inital heal and overhealing in theory shouldnt massively swing anyways
+   */
+  public getEchoOfLightHealingAttrib(eolEvent: HealEvent) {
+    const currentMastery = this.statTracker.currentMasteryPercentage * this.masteryScaler;
+    const initialHeal =
+      currentMastery * (eolEvent.amount + (eolEvent.absorbed || 0) + (eolEvent.overheal || 0));
+    let eolTicksHeal = 0;
+    let eolTicksOverHeal = 0;
+
+    // blacklist non EOL heals
+    if (ABILITIES_THAT_DONT_TRIGGER_MASTERY.includes(eolEvent.ability.guid)) {
+      return 0;
+    }
+
+    // get average healing/overhealing over the next 6s of EOL
+    const eolHealEvents = GetRelatedEvents(eolEvent, ECHO_OF_LIGHT_ATTRIB_EVENT);
+    if (eolHealEvents.length > 0) {
+      const empty: HealEvent[] = []; //eslint....
+      Object.values(
```

Comment:
same notes here about unneeded `Object.values` / using `forEach` instead of `reduce`